### PR TITLE
JAMES-2503 Remove Guavate dependency

### PR DESCRIPTION
**Pull request by aduprat**

### CODE REVIEW

File: event-sourcing/event-sourcing-core/pom.xml [hunk 1]
```diff
@@ -59,6 +59,10 @@
             <artifactId>equalsverifier</artifactId>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>org.apache.commons</groupId>
+            <artifactId>commons-lang3</artifactId>
+        </dependency>
```
Comment (rouazana) [inline event-sourcing/event-sourcing-core/pom.xml]: why do you need it?

Comment (aduprat) [reply event-sourcing/event-sourcing-core/pom.xml]: It's needed here:
https://github.com/linagora/james-project/blob/1968a528f61ee0c16515177b99919825ab3e0597/event-sourcing/event-sourcing-core/src/main/java/org/apache/james/eventsourcing/EventBus.java#L27

Guavate is bringing Guava & Commons-lang3 so I had to add them when needed & not declared.

**Comment by trantienduchn** [comment]
```
[bc05138506437d3885ce6fcde2137af7dcbf05b1]      but: <java.lang.NoSuchMethodError: com.google.common.collect.ImmutableList.toImmutableList()Ljava/util/stream/Collector;> is a java.lang.NoSuchMethodError
```

```
[bc05138506437d3885ce6fcde2137af7dcbf05b1] [ERROR] Errors: 
[bc05138506437d3885ce6fcde2137af7dcbf05b1] [ERROR]   com.google.common.collect.ImmutableList.toImmutableList()Ljava/util/stream/Collector;
[bc05138506437d3885ce6fcde2137af7dcbf05b1] [ERROR] Rewrite Tables tests.address mapping should be retrieved when searching with a domain alias(Rewrite Tables tests)
[bc05138506437d3885ce6fcde2137af7dcbf05b1] [ERROR]   Run 1: com.google.common.collect.ImmutableList.toImmutableList()Ljava/util/stream/Collector;
[bc05138506437d3885ce6fcde2137af7dcbf05b1] [ERROR]   Run 2: com.google.common.collect.ImmutableList.toImmutableList()Ljava/util/stream/Collector;
[bc05138506437d3885ce6fcde2137af7dcbf05b1] [INFO] 
[bc05138506437d3885ce6fcde2137af7dcbf05b1] [ERROR]   com.google.common.collect.ImmutableList.toImmutableList()Ljava/util/stream/Collector;
[bc05138506437d3885ce6fcde2137af7dcbf05b1] [ERROR]   HBaseRecipientRewriteTableTest>AbstractRecipientRewriteTableTest.addForwardMappingShouldStore:302 » NoSuchMethod
[bc05138506437d3885ce6fcde2137af7dcbf05b1] [ERROR]   HBaseRecipientRewriteTableTest>AbstractRecipientRewriteTableTest.addGroupMappingShouldStore:334 » NoSuchMethod
[bc05138506437d3885ce6fcde2137af7dcbf05b1] [ERROR]   HBaseRecipientRewriteTableTest>AbstractRecipientRewriteTableTest.addMappingShouldNotThrowWhenMappingAlreadyExistsWithAnOtherType:288 » NoSuchMethod
[bc05138506437d3885ce6fcde2137af7dcbf05b1] [ERROR]   HBaseRecipientRewriteTableTest>AbstractRecipientRewriteTableTest.getAllMappingsShouldListAllEntries:120 » NoSuchMethod
[bc05138506437d3885ce6fcde2137af7dcbf05b1] [ERROR]   HBaseRecipientRewriteTableTest>AbstractRecipientRewriteTableTest.removeForwardMappingShouldDelete:316 » NoSuchMethod
[bc05138506437d3885ce6fcde2137af7dcbf05b1] [ERROR]   HBaseRecipientRewriteTableTest>AbstractRecipientRewriteTableTest.removeGroupMappingShouldDelete:348 » NoSuchMethod
[bc05138506437d3885ce6fcde2137af7dcbf05b1] [ERROR]   HBaseRecipientRewriteTableTest>AbstractRecipientRewriteTableTest.testAliasDomainMapping:254 » NoSuchMethod
[bc05138506437d3885ce6fcde2137af7dcbf05b1] [ERROR]   HBaseRecipientRewriteTableTest>AbstractRecipientRewriteTableTest.testRecursiveMapping:230 » NoSuchMethod
[bc05138506437d3885ce6fcde2137af7dcbf05b1] [ERROR]   HBaseRecipientRewriteTableTest>AbstractRecipientRewriteTableTest.testStoreAndGetMappings:73 » NoSuchMethod
[bc05138506437d3885ce6fcde2137af7dcbf05b1] [ERROR]   HBaseRecipientRewriteTableTest>AbstractRecipientRewriteTableTest.testStoreAndRetrieveAddressMapping:145 » NoSuchMethod
[bc05138506437d3885ce6fcde2137af7dcbf05b1] [ERROR]   HBaseRecipientRewriteTableTest>AbstractRecipientRewriteTableTest.testStoreAndRetrieveErrorMapping:169 » NoSuchMethod
[bc05138506437d3885ce6fcde2137af7dcbf05b1] [ERROR]   HBaseRecipientRewriteTableTest>AbstractRecipientRewriteTableTest.testStoreAndRetrieveRegexMapping:88 » NoSuchMethod
[bc05138506437d3885ce6fcde2137af7dcbf05b1] [ERROR]   HBaseRecipientRewriteTableTest>AbstractRecipientRewriteTableTest.testStoreAndRetrieveWildCardAddressMapping:196 » NoSuchMethod
[bc05138506437d3885ce6fcde2137af7dcbf05b1] [INFO] 
[bc05138506437d3885ce6fcde2137af7dcbf05b1] [ERROR] Tests run: 108, Failures: 1, Errors: 57, Skipped: 2
```

**Comment by aduprat** [comment]
```
[ERROR] Failures: 
[ERROR]   CassandraMailboxRecentDAOTest.getRecentMessageUidsInMailboxShouldNotTimeoutWhenOverPagingLimit:142 
Expected size:<6000> but was:<7000> in:
```
test this please

**Comment by aduprat** [comment]
Can't be achieved yet: HBase implementation needs an old version of Guava (13) at runtime, which doesn't include the immutable collectors.

**Pull request closed without merging**